test/clj/rems/test_browser.clj

```diff
@@ -1354,65 +1386,280 @@
+          (btu/fill-human (field :columns-0-label-fi) "Table column 0 (FI)")
+          (btu/fill-human (field :columns-0-label-sv) "Table column 0 (SV)")
+
+          (btu/scroll-and-click (field :add-column))
+          (btu/fill-human (field :columns-1-key) "table-column-1")
+          (btu/fill-human (field :columns-1-label-en) "Table column 1 (EN)")
+          (btu/fill-human (field :columns-1-label-fi) "Table column 1 (FI)")
+          (btu/fill-human (field :columns-1-label-sv) "Table column 1 (SV)")
+
+          (btu/scroll-and-click (field :add-column))
+          (btu/fill-human (field :columns-2-key) "table-column-2")
+          (btu/fill-human (field :columns-2-label-en) "Table column 2 (EN)")
+          (btu/fill-human (field :columns-2-label-fi) "Table column 2 (FI)")
+          (btu/fill-human (field :columns-2-label-sv) "Table column 2 (SV)")
+
+          (let [idx (context-field-index :create-form-field/table)]
+            (is (= (->> (btu/query-all [{:id (str "container-form-1-field-fld" idx)}
+                                        {:css "table > thead > tr > th"}])
+                        (map btu/value-of-el)
+                        (remove str/blank?))
+                   ["Table column 0 (EN)"
+                    "Table column 1 (EN)"
+                    "Table column 2 (EN)"]))))
+
+        (testing "create date field"
+          (let [idx (create-context-field! :create-form-field/date)]
+            (btu/scroll-and-click-el (last (btu/query-all {:class :add-form-field})))
+            (btu/scroll-and-click (field :type-date))
+            (is (btu/eventually-visible? {:id (str "form-1-field-fld" idx)
+                                          :tag :input
+                                          :css "[type=date]"}))
+
+            (btu/fill-human (field :title-en) "Date field (EN)")
+            (btu/fill-human (field :title-fi) "Date field (FI)")
+            (btu/fill-human (field :title-sv) "Date field (SV)")
+
+            (btu/scroll-and-click (field :info-text-more-link))
+            (is (btu/eventually-visible? (field :info-text-en)))
+            (btu/fill-human (field :info-text-en) "Info text (EN)")
+            (btu/fill-human (field :info-text-fi) "Info text (FI)")
+            (btu/fill-human (field :info-text-sv) "Info text (SV)")
+            (is (btu/eventually-visible? [{:id (str "container-form-1-field-fld" idx)}
+                                          {:tag :button :fn/has-class "info-button"}]))
+
+            (btu/scroll-and-click [{:id (str "container-form-1-field-fld" idx)}
+                                   {:tag :button :fn/has-class "info-button"}])
+            (is (btu/eventually-visible? {:id (str "form-1-field-fld" idx "-collapse")}))
+            (is (= (btu/value-of {:id (str "form-1-field-fld" idx "-collapse")})
+                   "Info text (EN)"))))
+
+        (testing "create email address field"
```

Comment:
`email`, `phone` and `ip address` fields are similar text input. i think these should be tested by triggering validation but that does not show up in the preview currently

Reply:
as discussed, validation currently not implemented in preview and thus can be disregarded so far

Reply:
We have tests that check the types in the application so I think that's enough.